lib/nostrum/cache/guild/guild_server.ex

```diff
@@ -442,4 +447,14 @@ defmodule Nostrum.Cache.Guild.GuildServer do
     {:reply, {guild_id, old_emojis, emojis}, %{state | emojis: emojis}}
   end
 
+  def handle_call({:update, :voice_state, guild_id, voice_state}, _from, state) do
+    old_voice_states = state.voice_states
+    new_voice_states =
+      [
+        voice_state |
+        state.voice_states
+        |> Enum.reject(fn(map) -> map.user_id == voice_state.user_id end)
+      ]
+    {:reply, {guild_id, old_voice_states, new_voice_states}, %{state | voice_states: new_voice_states}}
```

Comment:
My line of thinking is that if people need to see the voice stats of others, they'll be able to fetch the guild and retrieve it from there. Since the number of voice states will be fewer than other lists (such as presences), it may not be a big deal to dispatch whole lists. 

Ultimately I'll leave the decision up to you. If later on down the road whatever choice made here proves to be the incorrect choice, it can always be changed!